router/cmd/plan_generator.go

```diff
@@ -51,8 +59,46 @@ func PlanGenerator(args []string) {
 	)
 	defer stop()
 
-	err := plan_generator.PlanGenerator(ctxNotify, cfg)
+	logLevel, err := logging.ZapLogLevelFromString(cfg.LogLevel)
 	if err != nil {
-		log.Fatalf("Error during command plan-generator: %s", err)
+		log.Fatalf("Could not parse log level: %s", err)
+	}
+
+	logger := logging.New(false, false, logLevel).
+		With(
+			zap.String("service", "@wundergraph/query-plan"),
+			zap.String("service_version", core.Version),
+		)
+	cfg.Logger = logger
+
+	// Automatically set GOMAXPROCS to avoid CPU throttling on containerized environments
+	_, err = maxprocs.Set(maxprocs.Logger(func(msg string, args ...interface{}) {
+		logger.Info(fmt.Sprintf(msg, args...))
+	}))
+	if err != nil {
+		logger.Fatal(fmt.Sprintf("could not set max GOMAXPROCS: %s", err.Error()))
```

Comment:
You should be able to use the zap logger semantics by default, without needing to use fmt.Sprintf, eg :-
```
logger.Fatal("Could not start router", zap.Error(err))
```